cfme/cloud/host_aggregates.py

```diff
@@ -40,29 +85,101 @@ class my_filters(Accordion):  # noqa
+            self.in_host_aggregates and
+            self.entities.title.text == obj.expected_details_title and
+            self.entities.breadcrumb.active_location == obj.expected_details_breadcrumb and
+            self.entities.relationships.get_text_of('Cloud Provider') == obj.provider.name
+        )
+
+    toolbar = HostAggregatesDetailsToolBar()
+    sidebar = HostAggregatesDetailsAccordion()
+    entities = HostAggregatesDetailsEntities()
 
 
 @attr.s
-class HostAggregates(BaseEntity):
-    pass
+class HostAggregates(BaseEntity, Taggable):
+    """
+    Host Aggregates class to support navigation
+    """
+    _param_name = "HostAggregate"
+
+    name = attr.ib()
+    provider = attr.ib()
+    ram = attr.ib(default=None)
+    vcpus = attr.ib(default=None)
+    disk = attr.ib(default=None)
+    swap = attr.ib(default=None)
+    rxtx = attr.ib(default=None)
+    is_public = attr.ib(default=True)
+    tenant = attr.ib(default=None)
+
+    def refresh(self):
+        """Refresh provider relationships and browser"""
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def instance_count(self):
+        """ number of instances using host aggregates.
+
+        Returns:
+            :py:class:`int` instance count.
+        """
+        view = navigate_to(self, 'Details')
+        return int(view.entities.relationships.get_text_of('Instances'))
 
 
 @attr.s
-class HostAggregatesCollection(BaseCollection):
+class HostAggregatesCollection(BaseCollection, TaggableCollection):
     ENTITY = HostAggregates
 
+    def all(self):
+        provider = self.filters.get('provider')  # None if no filter, need for entity instantiation
+        view = navigate_to(self, 'All')
+        result = []
+        flavors = view.entities.get_all(surf_pages=True)
+        for flavor in flavors:
+            if provider is not None:
+                if flavor.data['cloud_provider'] == provider.name:
+                    entity = self.instantiate(flavor.data['name'], provider)
+            else:
+                entity = self.instantiate(flavor.data['name'],
+                                          get_crud_by_name(flavor.data['cloud_provider']))
+            result.append(entity)
+        return result
+
 
 @navigator.register(HostAggregatesCollection, 'All')
-class All(CFMENavigateStep):
-    VIEW = HostAggregatesDefaultView
+class HostAggregatesAll(CFMENavigateStep):
+    VIEW = HostAggregatesAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self, *args, **kwargs):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Host Aggregates')
+
+
+@navigator.register(HostAggregates, 'Details')
+class HostAggregatesDetails(CFMENavigateStep):
+    VIEW = HostAggregatesDetailsView
+    prerequisite = NavigateToAttribute('parent', 'All')
+
+    def step(self, *args, **kwargs):
+        self.prerequisite_view.toolbar.view_selector.select('List View')
```

Comment:
Note this is unnecessary when using BaseEntities - get_entity will operate on list, grid, or tile view.